cli/cmd/summary.go

```diff
@@ -62,3 +73,87 @@ func checkFilePath(filename string) error {
 	dirPath := filepath.Dir(filename)
 	return os.MkdirAll(dirPath, 0755)
 }
+
+func generateSummaryTable() *schema.Table {
+	tableName := "cloudquery_sync_summaries"
+	t := schema.Tables{{
+		Name: tableName,
+		Transform: transformers.TransformWithStruct(
+			&syncSummary{},
+		),
+		Columns: []schema.Column{},
+	}}
+	if err := transformers.TransformTables(t); err != nil {
+		panic(err)
+	}
+	return t[0]
+}
+
+func sendSummary(writeClients []plugin.Plugin_WriteClient, destinationSpecs []specs.Destination, destinationsClients []*managedplugin.Client, destinationTransformers []*transformer.RecordTransformer, summary *syncSummary, noMigrate bool) error {
+	summaryTable := generateSummaryTable()
+	summaryTableSchema := summaryTable.ToArrowSchema()
+
+	defaultCaser := caser.New()
+	for i := range destinationsClients {
```

Comment:
So this will send all summaries to all configured destinations correct?
Does it make sense to have it multiple destinations? Or a destination should have only it's "own" summary?

Reply:
I think it is implemented that a destination will only get the information about its self. In the for loop where we send the data to the destination, we set all of the destination specific information so it only ever gets information about itself.

If there is something I am missing let me know!

Reply:
Ok you're right each destination will get its own information, but since [we're looping over the summaries](https://github.com/cloudquery/cloudquery/pull/17842/files#diff-ee974bad39e59893cee87960039e8c360d83713c82eefb050fdb5eb4ea22d47dR380) each summary will be sent multiple times to the same destination, for example with the following configuration:

```yaml
kind: source
spec:
  name: "gcp"
  path: "cloudquery/gcp"
  registry: "cloudquery"
  version: "v12.7.0"
  tables: ["gcp_storage_buckets"]
  spec:
    project_ids: ["cq-playground"]
  destinations: ["file", "postgresql"]
---
kind: destination
spec:
  name: "file"
  path: "cloudquery/file"
  version: "v5.0.0"
  write_mode: "append"
  send_sync_summary: true
  spec:
    path: "examples/files/{{TABLE}}/{{UUID}}.{{FORMAT}}"
    format: "json"
---
kind: destination
spec:
  name: "postgresql"
  path: "cloudquery/postgresql"
  version: "v8.0.4"
  send_sync_summary: true
  spec:
    connection_string: "postgresql://postgres:pass@localhost:5432/postgres?sslmode=disable"
```

I got 2 files with the exact same summary under `examples/files/cloudquery_sync_summaries` and two entries in the Postgres summary table:
![image](https://github.com/cloudquery/cloudquery/assets/26760571/2c8f3ad1-da31-41c0-a74a-ce989b5e7a15)

Reply:
Also looks like `cli_version`, `cq_sync_id` and `cq_sync_time` are missing

Reply:
Fixed